target/linux/econet/patches-6.12/101-econet-timer-add-en7528-support.patch

```diff
@@ -0,0 +1,289 @@
+--- a/arch/mips/econet/Kconfig
++++ b/arch/mips/econet/Kconfig
+@@ -29,6 +29,7 @@ choice
+ 		bool "EN7528 family"
+ 		select COMMON_CLK
+ 		select CPU_LITTLE_ENDIAN
++		select ECONET_EN7528_TIMER
+ 		select IRQ_MIPS_CPU
+ 		select MIPS_CPU_SCACHE
+ 		select MIPS_GIC
+--- a/drivers/clocksource/Kconfig
++++ b/drivers/clocksource/Kconfig
+@@ -81,6 +81,16 @@ config ECONET_EN751221_TIMER
+ 	help
+ 	  Support for CPU timer found on EcoNet MIPS based SoCs.
+ 
++config ECONET_EN7528_TIMER
++	bool "EcoNet EN7528 CPU Timer" if COMPILE_TEST
++	depends on HAS_IOMEM
++	depends on MIPS || COMPILE_TEST
++	select CLKSRC_MMIO
++	select TIMER_OF
++	help
++	  Support for external CPU timer found on EcoNet EN7528 SoC.
++	  This timer uses separate GIC interrupts for each VPE.
++
+ config FTTMR010_TIMER
+ 	bool "Faraday Technology timer driver" if COMPILE_TEST
+ 	depends on HAS_IOMEM
+--- a/drivers/clocksource/Makefile
++++ b/drivers/clocksource/Makefile
+@@ -18,6 +18,7 @@ obj-$(CONFIG_CLKSRC_MMIO)	+= mmio.o
+ obj-$(CONFIG_DAVINCI_TIMER)	+= timer-davinci.o
+ obj-$(CONFIG_DIGICOLOR_TIMER)	+= timer-digicolor.o
+ obj-$(CONFIG_ECONET_EN751221_TIMER)	+= timer-econet-en751221.o
++obj-$(CONFIG_ECONET_EN7528_TIMER)	+= timer-econet-en7528.o
+ obj-$(CONFIG_OMAP_DM_TIMER)	+= timer-ti-dm.o
+ obj-$(CONFIG_OMAP_DM_SYSTIMER)	+= timer-ti-dm-systimer.o
+ obj-$(CONFIG_DW_APB_TIMER)	+= dw_apb_timer.o
+--- /dev/null
++++ b/drivers/clocksource/timer-econet-en7528.c
+@@ -0,0 +1,247 @@
++// SPDX-License-Identifier: GPL-2.0
++/*
++ * Timer present on EcoNet EN7528 MIPS based SoCs.
++ *
++ * Unlike EN751221 which uses a single percpu IRQ, EN7528 uses separate
```

Comment:
So you suppose there is a way to make the timer able to do both modes? I assume everything with GIC is going to be like this, so that includes also the EN751627, and "there's no code like no code" so one timer is better than two.

Reply:
let me try. What do you suggest? as part of this PR or separate? Also one question, for this PR to be merged, is it needed to be in upstream?